### ISSUE TITLE
NullPointerException when matching  array index from property that does not exist

### ISSUE THREAD

**Issue by robmartin0** [lukas-krecan/JsonUnit#94]
```
    @Test
    public void nullPointerExceptionTest() {
        String message = "{"
                + "     \"properties\":{"
                + "         \"attr\":\"123\""
                + "     }"
                + "}";
        String path = "properties.another[0]";
        String expected = "VALUE";
        final ConfigurableJsonMatcher<Object> jsonMatcher = jsonPartEquals(path, expected);
        jsonMatcher.matches(message);
    }
```

throws a NullPointerException:
`java.lang.NullPointerException
	at net.javacrumbs.jsonunit.core.internal.Diff.compare(Diff.java:91)
	at net.javacrumbs.jsonunit.core.internal.Diff.similar(Diff.java:564)
	at net.javacrumbs.jsonunit.JsonMatchers$JsonPartMatcher.matches(JsonMatchers.java:158)`

**Comment by lukas-krecan** [comment]
Thanks for reporting it, will fix in in the next release

**Comment by lukas-krecan** [comment]
Released as 1.28.1
